dingo/pipe/data_generation.py

```diff
@@ -162,47 +194,187 @@ def __init__(self, args, unknown_args, create_data=True):
+                    else None
+                )
+                self.strain_data_list.append(
+                    injection_generator.injection(
+                        self.injection_dict,
+                        seed=seed,
+                    )
+                )
+            # useful for computing optimal SNR
+            strain_data = self.strain_data_list[0]
+        else:
+            strain_data = injection_generator.injection(self.injection_dict)
+            self.strain_data = strain_data
+
+        # Compute optimal SNR
+        rho_opt_ifos, rho_opt = self.compute_optimal_snr(
+            strain_data, injection_generator.data_domain
+        )
+        logger.info(f"Network optimal SNR of injection: {rho_opt}")
+        logger.info(f"Detector optimal SNRs of injection: {rho_opt_ifos}")
+
+    def compute_optimal_snr(self, strain_data, data_domain):
+        """Compute network optimal signal-to-noise ratio for the first injected strain"""
+        mu = strain_data["waveform"]
+        asds = strain_data["asds"]
+        delta_f = data_domain.delta_f
+        noise_std = data_domain.noise_std
+
+        # In the inner products below explicitly divide by the window factor
+        window_factor = 4 * delta_f * noise_std**2
+
+        # optimal network SNR
+        kappa2_list = [
+            inner_product(
+                mu_ifo, mu_ifo, delta_f=delta_f, psd=window_factor * asd_ifo**2
+            )
+            for mu_ifo, asd_ifo in zip(mu.values(), asds.values())
+        ]
+        rho_opt = np.sqrt(sum(kappa2_list))
+        rho_opt_ifos = np.sqrt(kappa2_list)
+
+        return rho_opt_ifos, rho_opt
 
-            strain = ifo.strain_data.frequency_domain_strain
+    def create_data(self, args):
+        super().create_data(args)
+
+    def create_data(self, args):
+        super().create_data(args)
+        # check if there are nan's in the asd, if there are shift the detector segment used to generate the psd to an earlier time
+        for ifo in self.interferometers:
             frequency_array = ifo.strain_data.frequency_array
             asd = ifo.power_spectral_density.get_amplitude_spectral_density_array(
                 frequency_array
             )
 
-            # These arrays extend up to self.sampling_frequency. Truncate them to
-            # self.maximum_frequency, and also set the asd to 1.0 below
-            # self.minimum_frequency.
-            domain = FrequencyDomain(
-                f_min=self.minimum_frequency,
-                f_max=self.maximum_frequency,
-                delta_f=1 / self.duration,
-            )
-            strain = domain.update_data(strain)
-            asd = domain.update_data(asd, low_value=1.0)
-
-            # Dingo expects data to have trigger time 0, so we apply a cyclic time shift
-            # by the post-trigger duration.
-            strain = domain.time_translate_data(strain, self.post_trigger_duration)
+            if np.max(np.isnan(asd)):
```

Comment:
Easier to understand:
```suggestion
            if np.any(np.isnan(asd)):
```

Reply:
done